pkg/activator/handler/probe_handler_test.go

```diff
@@ -110,3 +110,29 @@ func mapToHeader(m map[string]string) http.Header {
 	}
 	return h
 }
+
+func BenchmarkProbeHandler(b *testing.B) {
+	baseHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	})
+	handler := ProbeHandler{NextHandler: baseHandler}
+	req := httptest.NewRequest(http.MethodGet, "http://example.com", nil)
+	req.Header.Set(network.ProbeHeaderName, activator.Name)
+	test := func() {
+		resp := httptest.NewRecorder()
```

Comment:
Not sure if this is causing the memory allocation. Can we try creating a new response write only for the parallel test, not the sequential one?

Reply:
Yeah this can probably move before the respective `for` loop in each of the `Run` calls. Ditch the `test` function then and just call ServeHTTP directy.